tests/app/data_model/test_database_setup.py

```diff
@@ -47,11 +52,32 @@ def test_database_retry_success_after_attempts(self):
             setup.side_effect = iter([ConnectionRefusedError, ConnectionRefusedError, 'success'])
 
             # When I call the create with retry function
-            database = Database("sqlite://", 3, 6)
+            database = Database(database_driver="sqlite",
+                                database_name="",
+                                database_setup_attempts=3,
+                                database_setup_retry_delay=0)
 
             # pylint: disable=protected-access
             # Then a value is successfully returned
             self.assertEqual(database._db_session, 'success')
 
             # And the setup was called three times before succeeding
             self.assertEqual(setup.call_count, 3)
+
+    def test_database_url_creation_for_non_sqlite(self):
+        # Given the database setup fails twice then succeeds
+        with patch('app.data_model.database.Database._create_session_and_engine'):
+
+            # When I call the create with retry function
+            database = Database(database_driver="postgresql",
+                                database_host="localhost",
+                                database_port=5432,
+                                username="testUsername",
+                                password="testPassword",
+                                database_name="digitaleqrds",
+                                database_setup_attempts=1,
+                                database_setup_retry_delay=0)
+
+            # pylint: disable=protected-access
+            # Then a value is successfully returned
+            self.assertEqual(database._database_url, 'postgresql://testUsername:testPassword@localhost:5432/digitaleqrds')
```

Comment:
If the database url generation was factored out into it's own method, we wouldn't need to mock the database to test it.